app/models/statistics_announcement_date.rb

```diff
@@ -4,7 +4,7 @@ class StatisticsAnnouncementDate < ApplicationRecord
   belongs_to :statistics_announcement, touch: true
   belongs_to :creator, class_name: "User"
 
-  after_save :update_statistics_announcement_current_release_date
+  after_save :update_statistics_announcement_current_release_date, if: -> { statistics_announcement.present? }
```

Comment:
It seems a bit dodgy that `statistics_announcement` might not be present 🤔 I would have expected a validation rule to ensure it's always there.

But I don't think that's something to be solved in this PR. So this looks good 👍🏻

Reply:
yea agreed. Editions get around this by having a before_save that creates a document. It's probs worth exploring something similar for this down the line.